src/cider/nrepl/middleware/track_state.clj

```diff
@@ -281,21 +345,32 @@
          all-namespaces (if cljs
                           (vals (cljs-ana/all-ns cljs))
                           (all-ns))
-         project-ns-map (fast-reduce (fn [acc ns]
-                                       (let [name (ns-name-fn ns)]
-                                         (if (jar-ns-fn name)
-                                           acc ;; Remove all jar namespaces.
-                                           (assoc! acc name (ns-as-map ns all-namespaces)))))
-                                     all-namespaces)
+         ns-as-map-fn (memoize ns-as-map)
+         project-ns-map (claypoole/with-shutdown! [pool (claypoole/threadpool (if (System/getenv "CI")
+                                                                                1
+                                                                                4))]
+                          (->> all-namespaces
+                               (claypoole/pmap pool
```

Comment:
The parallelism is necessary because we're reading files, for reading forms.

4 seems a good value (it's an IO bound problem, but we don't want to create a lot of threads).